guides/openai-assistant-py/openai_assistant/guide.md

```diff
@@ -0,0 +1,497 @@
+We start the sandbox and configure the AI assistant in git. We log the user to GitHub via the authentication token.
+We assign the user's task to the assistant. Then we create a thread, send messages to the thread, and finally run the thread.
+
+> Here, we are using the OpenAI's concept of threads, messages and runs.
+
+```python
+def main():
+    sandbox = Sandbox(
+        on_stderr=lambda message: print("[Sandbox stderr]", message),
+        on_stdout=lambda message: print("[Sandbox stdout]", message),
+    )
+    sandbox.add_action(read_file).add_action(save_code_to_file).add_action(
+        list_files
+    ).add_action(commit_and_push).add_action(write_to_file)
+
+    # Identify AI developer in git
+    sandbox.process.start_and_wait(
+        "git config --global user.email 'ai-developer@email.com'"
+    )
+    sandbox.process.start_and_wait("git config --global user.name 'AI Developer'")
+
+    user_gh_token = prompt_user_for_auth()
+    # Log in to github
+    print("Logging you into your GitHub...")
+    proc = sandbox.process.start_and_wait(
+        f"echo {user_gh_token} | gh auth login --with-token"
+    )
+    if proc.exit_code != 0:
+        print("Error: Unable to log into github")
+        print(proc.stderr)
+        print(proc.stdout)
+        exit(1)
+    # Check that user is logged into github
+    proc = sandbox.process.start_and_wait("gh auth status")
+    if proc.exit_code != 0:
+        print("Error: Unable to log into github")
+        print(proc.stderr)
+        print(proc.stdout)
+        exit(1)
+    # Check that user is authenticated
+    proc = sandbox.process.start_and_wait("gh auth setup-git")
+    if proc.exit_code != 0:
+        print("Error: Unable to se up Git auth with GitHub")
+        print(proc.stderr)
+        print(proc.stdout)
+        exit(1)
+
+    repo_url = prompt_user_for_github_repo()
+    user_task = prompt_user_for_task(repo_url)
+
+    git_clone_proc = sandbox.process.start_and_wait(
+        f"git clone {repo_url} {repo_directory}"
+    )
+    if git_clone_proc.exit_code != 0:
+        print("Error: Unable to clone the repository")
+        exit(1)
+
+    thread = client.beta.threads.create(
+        messages=[
+            {
+                "role": "user",
+                "content": f"Carefully plan this task and start working on it: {user_task} in the {repo_url} repository",
+            },
+        ],
+    )
+
+    run = client.beta.threads.runs.create(
+        thread_id=thread.id, assistant_id=assistant.id
+    )
+```
+
+Still  inside the main function, we print the assistant's process to the terminal. Each time it chooses to use one of the actions, we see in the terminal how it uses the action and whether it ends with success or error.
+
+![Assistants status](005.png "A schema of assistant's run steps")
+
+> Note: If the assistant is "queued", we are waiting for OpenAI's processes.
+
+At the end, we close the sandbox.
+
+```
+while True:
```

Comment:
the indent isn't right. You can either do this
```suggestion
    while True:
```
or move everything below 1 tab to the left